### PR TITLE
Add support for grid charts

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [x] documentation updated

**Squash commit message**

Add support for getting data for `sn-grid-chart`.

**Info**

Saving tree data pages has been added in this PR, however two steps still left to be able to support `select` in grid chart

* Add support for hypercube tree mode in `getpossible` part of select function.
* Add support for sending `SelectPivotCells` select request using the saved tree data pages.

The `allvalues` part of getting data might need re-evaluation in case we add more objects based on tree data in the future. Current will send correct requests for grid charts.

### CODE REVIEW

File: session/objecthandling.go [hunk 1]
```diff
@@ -554,10 +560,66 @@ func UpdateObjectHyperCubeContinuousDataAsync(sessionState *State, actionState *
...
+
+		for i, dim := range dimInfo {
+			height := dim.Cardinal
+			if height > requestDef.Height {
+				height = requestDef.Height
+			}
+
+			allValues := true
+			if i == 0 {
+				allValues = false
```
Comment (SeBBBe) [inline session/objecthandling.go:595]: allValues := i != 0

**Review by SeBBBe — APPROVED** [review]
LGTM!

Is the lack of selection support the reason you were trying to take a look at the grid chart source code? If it's something that seems feasible to do we should open an issue on it for the future.

File: session/objecthandling.go (hunk 1 above)
Comment (DnlLrssn) [reply session/objecthandling.go:595]: ah true, a bit cleaner

Comment (DnlLrssn) [reply session/objecthandling.go:595]: No wanted the code for more generic support for treedata and specifically `allValues` field. This is correct for grid chart which is the only such object right now.

For the select I know what to do now, it's just too much to change right now and want to prioritize other things.

Comment (SeBBBe) [reply session/objecthandling.go]: Perfect! That's fine; getting this level of support in the gopherciser is very useful not least for the app-evaluator. I agree that the select part is less important.

File: session/objecthandling.go [hunk 2]
```diff
@@ -387,6 +389,7 @@ func UpdateObjectHyperCubeReducedDataAsync(sessionState *State, actionState *act
 	}, actionState, true, fmt.Sprintf("Failed to update object hypercube reduced data for object<%s>", gob.GenericId))
 }
 
+// UpdateObjectHyperCubeBinnedDataAsync sed get hypercube binned data request and update saved hypercube
```
Comment (atluq) [inline session/objecthandling.go:392]: sed --> send

**Review by atluq — APPROVED** [review]
LGTM!

File: session/objecthandling.go (hunk 2 above)
Comment (DnlLrssn) [reply session/objecthandling.go:392]: `sed -i 's/sed/send/g' objecthandling.go`

**Pull request merged**